### PR TITLE
fix: multiple config compilation error (rebase)

### LINKED ISSUE #887: [BUG]: Multiple config throws error calling tap on undefined in webpack-cli
# Bug report

**What is the current behavior?**
When running webpack with `--info-verbosity verbose` and multiple configs the following error is thrown.  The MultiCompiler does not define a beforeRun hook.

```node_modules/webpack-cli/bin/cli.js:281
    compiler.hooks.beforeRun.tap("WebpackInfo", compilation => {
					        ^

TypeError: Cannot read property 'tap' of undefined
    at processOptions (node_modules/webpack-cli/bin/cli.js:281:31)
    at yargs.parse (node_modules/webpack-cli/bin/cli.js:373:3)
    at Object.parse (node_modules/webpack-cli/node_modules/yargs/yargs.js:567:18)
    at node_modules/webpack-cli/bin/cli.js:49:8
    at Object.<anonymous> (node_modules/webpack-cli/bin/cli.js:375:3)
    at Module._compile (internal/modules/cjs/loader.js:701:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:712:10)
    at Module.load (internal/modules/cjs/loader.js:600:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:539:12)
    at Function.Module._load (internal/modules/cjs/loader.js:531:3)
    at Module.require (internal/modules/cjs/loader.js:637:17)
    at require (internal/modules/cjs/helpers.js:22:18)
    at Object.<anonymous> (node_modules/webpack/bin/webpack.js:156:2)
    at Module._compile (internal/modules/cjs/loader.js:701:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:712:10)
    at Module.load (internal/modules/cjs/loader.js:600:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:539:12)
    at Function.Module._load (internal/modules/cjs/loader.js:531:3)
    at Function.Module.runMain (internal/modules/cjs/loader.js:754:12)
    at startup (internal/bootstrap/node.js:283:19)
    at bootstrapNodeJSCore (internal/bootstrap/node.js:622:3)
```

**If the current behavior is a bug, please provide the steps to reproduce.**
Create a multiple config webpack and run webpack with --info-verbosity verbose.

**What is the expected behavior?**
MultiCompiler should define a beforeRun hook.

**Other relevant information:**
webpack version:  4.32.1
Node.js version: v10.15.3
Operating System: MacOS 10.14.4
Additional tools:

### PR DESCRIPTION
**Pull request by aman29271**
**What kind of change does this PR introduce?**
A bugfix which closes #887 

**Did you add tests for your changes?**
Yes
**If relevant, did you update the documentation?**
Nope

**Does this PR introduce a breaking change?**
No

**Other information**

### CODE REVIEW

**Comment by DeadCoderAlive** [comment]
Do we have an ETA for the merge

File: test/binCases/entry/multiple-config/multiple-config.test.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
+"use strict";
+
+const { run, extractSummary } = require("../../../testUtils");
+
+test("multiple-config", () => {
+	const { stdout, code, stderr } = run(__dirname, [
+		"--config",
+		"./webpack.config.js",
+		"--output-file-name",
```
Comment (snitin315) [inline test/binCases/entry/multiple-config/multiple-config.test.js:9]:
```suggestion
		"--output-filename",
```

Comment (snitin315) [inline test/binCases/entry/multiple-config/multiple-config.test.js:9]:
```suggestion
		"--output-filename",
```

Comment (snitin315) [reply test/binCases/entry/multiple-config/multiple-config.test.js:9]: it should be `--output-filename`

**Comment by snitin315** [comment]
Also, can you please add a test case for `--info-verbosity verbose` with `multiple configs` as per the Issue, So that we can ensure we don't have the same problem anymore?

**Comment by aman29271** [comment]
@snitin315 Thanks for pointing out. Tests has been added.

File: test/binCases/entry/multiple-config/multiple-config.test.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+		"--mode",
+		"production"
+	]);
+
+	const summary = extractSummary(stdout);
+
+	expect(code).toBe(0);
+	expect(code).toBeDefined();
+	expect(code).not.toBeNull();
+	expect(summary).toEqual(expect.anything());
```
Comment (evilebottnawi) [inline test/binCases/entry/multiple-config/multiple-config.test.js:20]: Bad idea to do it, no need test on anything

Comment (aman29271) [reply test/binCases/entry/multiple-config/multiple-config.test.js:20]: @evilebottnawi should I remove it? :thinking:

**Comment by evilebottnawi** [comment]
Fixed in master, anyway big thanks for PR

**Pull request closed without merging**